src/en/03_user_guide.md

```diff
@@ -1,23 +1,156 @@
+
+![Display of the monitor unit creation form](./images/monitor_unit.png)
+
+**Backends:** The backends tab displays a table with information about ports, addresses, events, and database activity.
+
+---
 
 ## Snippets
 
-## Command history
+The snippets panel can be accessed by clicking on the snippet symbol on the right side menu. You may write code on the text editor and save it. The indentation feature will automatically rewrite the SQL in a pattern.
+
+![Image displaying the snippets panel](./images/snippet.png)
+
+To use a saved snippet, open a query tap and right click on the text editor. Here you may use, overwrite, or create a snippet.
+
+![Image showing how to use snippets on the query tab](./images/snippet1.png)
+
+---
+
+## Command History
+
+The command history is available in the query and console tabs. It displays a list of previously executed queries/commands in a given time range. To copy a query from the history, right click and select `Copy Content To Query Tab`.
+
+![Image showing the query history](./images/history.png)
+
+---
 
 ## Explain
 
-## Code autocompletion
+The **Explain** and **Explain Analyze** features provide a visualization of the PostgreSQL execution plan for a query. The Explain feature provides the visualization with cost information while the Explain Analyze feature provides extra information -- such as time, number of rows filtered, and buffers. 
+
+You may access the graph metrics by clicking on `Settings` at the top right. This allows to display information about cost, time, and rows. Next, you may click on a graph node to display more information about it. Finally, zoom in and out of the graph with the scroll wheel.
+
+![Screenshot for the explain analyze feature](./images/explain_analyze.png)
+
+---
+
+## Code Autocompletion
+
+To aid at typing speed, the application will suggest tables or indexes to autocomplete the query.
+
+![Image showing the code autocompletion](./images/autocomplete.png)
+
+---
+
+## PostgreSQL Configuration Management
+
+To access the server configuration for a PostgreSQL connection, right click on the server object on the DB entity tree and select `Server Configuration`.
+
+![Image with pointer clicking the Server Configuration button](./images/configuration.png)
+
+A new tab will open with the server configuration settings. You may search for a particular setting, choose a category, or revert a configuration with the `Config History` drop down menu. After making the desired changes to the configuration click `Apply` to save the configuration. A message will appear confirming the changes made and asking for an optional comment. Once committed, the new configuration should appear in the `Config History` drop down menu.
+
```

Comment:
It is important to mention that previous configuration state is automatically save to a snapshot and it can be restored back if needed.
Also, some configuration changes may require a server process reload, which should be done manually. Pgmanage will let user know that server is pending restart if at least one of altered settings require that.